server/index.js

```diff
@@ -8,18 +8,34 @@ app.use(parser.json());
 app.use(express.static(__dirname + '/../dist'));
 //get the names and pics of the products
 app.get('/products', function (req, res) {
-  res.send(models.getProducts());
+  models.getProducts((err, data) => {
+    if (err) {
+      throw err;
+    } else {
+      res.send(data);
+    }
+  });
 });
 
-//add product name, url, and username
+//add product name, url, username, like
 app.post('/products', function (req, res) {
-  models.saveProduct(req.body.productId, req.body.productItem, req.body.pictureUrl);
+  models.saveProduct(req.body.productId, req.body.productItem, req.body.pictureUrl, req.body.like);
+  res.end('done');
+});
+//update like of productId
+app.put('/products', function (req, res) {
+  models.updateProduct(req.body.productId, req.body.like);
   res.end('done');
```

Comment:
This is an asynchronous race condition, and the response is ended with `done` before the database has been successfully updated (or perhaps errors out); we should respond with an error status code/message if an error occurs.